core/src/replay_stage.rs

```diff
@@ -1171,12 +1174,247 @@ pub(crate) mod tests {
         system_transaction,
         transaction::TransactionError,
     };
-    use solana_vote_program::vote_state::VoteState;
+    use solana_stake_program::stake_state;
+    use solana_vote_program::vote_state;
+    use solana_vote_program::vote_state::{Vote, VoteState};
     use std::{
         fs::remove_dir_all,
         sync::{Arc, RwLock},
     };
 
+    #[test]
+    fn test_minority_fork_overcommit_attack() {
+        // Minority was leader for 9 slots consecutively, withheld votes on that
+        // until majority built 8 slots
+        //
+        // Minority     -> 3 -> 4 -> 5 -> 6 -> 7 -> 8 -> 9 -> 10 -> 11
+        // 0 -> 1 -> 2
+        // Majority     -> 12 -> 13 -> 14 -> 15 -> 16 -> 17 -> 18 -> 19
+        //
+
+        fn vote(bank: &Arc<Bank>, pubkey: &Pubkey, slot: Slot) {
+            let mut vote_account = bank.get_account(&pubkey).unwrap();
+            let mut vote_state = VoteState::from(&vote_account).unwrap();
+            vote_state.process_slot_vote_unchecked(slot);
+            vote_state.to(&mut vote_account).unwrap();
+            bank.store_account(&pubkey, &vote_account);
+        }
+
+        const BOOTSTRAP_LEADER: usize = 0;
+        const HONEST_NODE: usize = 1;
+        const MALICIOUS_NODE: usize = 2;
+
+        let neutral_path: Vec<u64> = vec![0, 1, 2];
+        let minority_fork: Vec<u64> = vec![3, 4, 5, 6, 7, 8, 9, 10, 11];
+        let majority_fork: Vec<u64> = vec![12, 13, 14, 15, 16, 17, 18, 19, 20];
```

Comment:
Might be good to encode why we chose an 8-length path here (maybe b/c it's the threshold depth?). So something like `(12..12 + VOTE_THRESHOLD_DEPTH).collect();`, and for the minority fork, `(3..3+VOTE_THRESHOLD_DEPTH + 1).collect();`